### PR TITLE
Stable threaded skeleton search

### PR DESCRIPTION
**Pull request by mschauer**

### CODE REVIEW

File: src/skeleton.jl [hunk 1]
```diff
@@ -39,22 +61,76 @@ function skeleton(g, I, par, kwargs)
...
+        d = d + 1
+        if isdone
+            return g, S
+        end
+    end
+end
+
+"""
+    skeleton_stable(n::Integer, I) -> g, S
+    skeleton_stable(g, I) -> g, S
```
Comment (mschauer) [inline src/skeleton.jl:90]: This is the interesting bit

**Comment by mschauer** [comment]
Timings versus pc-alg on 4 threads:

```
> system.time(skel.fit <- skeleton(suffStat = list(C = cor(dat), n = n),
+                                  indepTest = gaussCItest, ## (partial correlations)
+                                  alpha = 0.001, p = p,  method = "stable.fast"))#
   user  system elapsed 
 13.693   0.033  13.759 
> skel.fit
Object of class 'pcAlgo', from Call:
skeleton(suffStat = list(C = cor(dat), n = n), indepTest = gaussCItest, 
    alpha = 0.001, p = p, method = "stable.fast")
Number of undirected edges:  254 
Number of directed edges:    0 
Total number of edges:       254 
> 
```
versus 
```
julia> h, s = @time CausalInference.skeleton(d, gausscitest, (cor(X), n), -quantile(Normal(), alpha/2)); h
 13.371163 seconds (42.97 M allocations: 10.909 GiB, 19.89% gc time)
{64, 254} undirected simple Int64 graph
```

**Comment by mwien** [comment]
Nice! Before we were slower?

**Comment by mschauer** [comment]
Yes, a bit slower as the C implementation "stable.fast" contained in `pcalg`, by a factor of 2 which is okay (and much faster than the native implementation "stable" of course which is the default in `pcalg` )

**Comment by mwien** [comment]
What if you pass numCores = 4 in the pcalg call?

**Comment by mschauer** [comment]
Hm, you are right, but not much happens:
```
> system.time(skel.fit <- skeleton(suffStat = list(C = cor(dat), n = n),
+                                  indepTest = gaussCItest, ## (partial correlations)
+                                  alpha = 0.001, p = p,  method = "stable.fast", numCores = 4))#, labels = V
   user  system elapsed 
 13.838   0.063  14.031 
> 
```

**Comment by mwien** [comment]
Hmm weird...
But not our problem I guess...

**Comment by mschauer** [comment]
That is more tricky on a second note. Not sure if I even have OpenMP support. https://stackoverflow.com/questions/5008092/setting-up-configure-for-openmp-in-r

**Pull request merged**